producer/writer/consumer_writer.go

```diff
@@ -200,9 +200,16 @@ func (w *consumerWriterImpl) Init() {
 	}()
 }
 
-func (w *consumerWriterImpl) resetConnectionUntilClose() {
+func (w *consumerWriterImpl) cleanupConnectionUntilClose() {
+	flushTicker := time.NewTicker(w.connOpts.FlushInterval())
```

Comment:
Do you want to use a separate goroutine for this logic? Not sure if putting both flush and reseting connection in one goroutine is a great idea, makes it hard to reason about.

Reply:
The reason to put it here is that when you need to reset the connection that basically means the connection is broken and you don't need to flush, and vice versa. I guess a new go routine could make it cleaner.